test/test_dna_text.py

```diff
@@ -0,0 +1,142 @@
+from unittest import main, TestCase
+import tempfile
+import os
+from lxml import etree
+
+from readalongs.log import LOGGER
+from readalongs.app import app
+from readalongs.text import tokenize_xml
+from readalongs.text.add_ids_to_xml import add_ids
+
+class TestDNAText(TestCase):
+    LOGGER.setLevel('DEBUG')
+    data_dir = os.path.join(os.path.dirname(__file__), 'data')
+    maxDiff = None
+
+    def setUp(self):
+        pass
+
+    def tearDown(self):
+        pass
+
+    def test_tok_all_words(self):
+        txt = """<document xml:lang="fra">
+<s>Bonjour! Comment ça va?</s>
+<s>Voici une deuxième phrase.</s>
+</document>"""
+        xml = etree.fromstring(txt)
+        tokenized = tokenize_xml.tokenize_xml(xml)
+        as_txt = etree.tounicode(tokenized)
+        #print(etree.tounicode(tokenized))
+
+        ref = """<document xml:lang="fra">
+<s><w>Bonjour</w>! <w>Comment</w> <w>ça</w> <w>va</w>?</s>
+<s><w>Voici</w> <w>une</w> <w>deuxième</w> <w>phrase</w>.</s>
+</document>"""
+        #print('as_txt="' + as_txt +'"')
+        #print('ref="' + ref +'"')
+        self.assertEqual(as_txt, ref)
+
+        with_ids = add_ids(tokenized)
+        ids_as_txt = etree.tounicode(with_ids)
+        #print('with ids="' + ids_as_txt + '"')
+        ref_with_ids = """<document xml:lang="fra">
+<s id="s0"><w id="s0w0">Bonjour</w>! <w id="s0w1">Comment</w> <w id="s0w2">ça</w> <w id="s0w3">va</w>?</s>
+<s id="s1"><w id="s1w0">Voici</w> <w id="s1w1">une</w> <w id="s1w2">deuxième</w> <w id="s1w3">phrase</w>.</s>
+</document>"""
+        self.assertEqual(ids_as_txt, ref_with_ids)
+
+    def test_tok_some_words(self):
+        txt = """<document xml:lang="fra">
+<p><s>Bonjour! Comment ça va?</s></p>
+<p do-not-align="true"><s>Bonjour! Comment ça va?</s></p>
+<s do-not-align="TRUE">Voici une deuxième phrase.</s>
+<s>Un <foo do-not-align="1">mot ou deux</foo> à exclure.</s>
```

Comment:
nvm, the question of omitting words is answered here. Maybe the template should be updated then as it only mentions `<s>` and `<p>` and `<div>` elements?

Reply:
Yeah, that's a good solution.